models/comment.py

```diff
@@ -17,6 +17,12 @@ def __init__(self, raw_comment=None, parent_comment_id=None):
     def __str__(self):
         return '{}: {}'.format(self.comment_author, self.comment_text)
 
+    def __eq__(self, other):
+        return self.comment_id == other.comment_id
+
+    def __hash__(self):
+        return hash(self.comment_id)
```

Comment:
what are those for?

Reply:
those are needed to make the set operations work. I needed to implement the __eq__ method to ensure correct equal checks. if you change the comparator you also need to change the hash function since you must ensure that to equal object have the same hash.